pkg/engine/virtualmachinescalesets.go

```diff
@@ -385,6 +389,10 @@ func CreateAgentVMSS(cs *api.ContainerService, profile *api.AgentPoolProfile) Vi
 	k8sConfig := orchProfile.KubernetesConfig
 	linuxProfile := cs.Properties.LinuxProfile
 
+	if k8sConfig != nil && k8sConfig.UseManagedIdentity && k8sConfig.UserAssignedID != "" {
+		dependencies = append(dependencies, "[concat('Microsoft.ManagedIdentity/userAssignedIdentities/', variables('userAssignedID'))]")
+	}
```

Comment:
I don't see where we deploy the identity in the ARM template. and if that's true, the ARM deployment will fail.

Reply:
found it. nvm.